### ISSUE TITLE
TPP threat to government owned open source IP

### ISSUE THREAD

**Issue by noisymime** [WhiteHouse/source-code-policy#30]
Article 14.17 of the Trans Pacific Partnership (TPP) deals with the treatment of source code and restricts what actions a member government may perform. 
Specifically:  “No Party shall require the transfer of, or access to, source code of software owned by a person of another Party, as a condition for the import, distribution, sale or use of such software, or of products containing such software, in its territory.”

Whilst the intent of this article was likely otherwise, the text as written will likely have an impact on the ability of governments to enforce their own copyrights on any Open Source product or service against a company or person from another TPP nation. 

Note that this is something that only applies to governments. Private organisations and individuals are not 'parties' in the context of the TPP, but governments and government agencies are. 

Research into this article indicates that none of the additional clauses would change the above. This includes 14.17.3(a) as licenses such as those commonly used in open source software (e.g. the GNU General Public License - GPL) are not considered ‘commercially negotiated contracts’. 

Has there been any consideration of this at all?

**Comment by steven-armstrong** [comment]
This proposal to have our government use open source GPL licenced software is perhaps one of the really great ideas of this administration, one that will foster innovation, inspiration and trust both among US citizen developers and globally.  This is a laudable major step in promoting transparency, and an inspired and courage act of public service, which will help instill and promote public trust and a feeling of being included, and facilitates a vast community of citizen developers to review code for potential bugs or security harms, as well as to provide new features and enhancements, all benefiting the institutions and agencies our citizen tax payers already funding government to the cost of hundreds of billions.  Starting to do this right now will be a great cost saver short-term and long-term, reducing duplication of efforts, unnecessary duplication of costs (and of course the duplicate bugs or failures of features), for generations to come.  I cannot imagine a greater long-term or more opportune bequest to American tax-payers (and the world) in this fast-moving Information Age, where Artificial Intelligence and Analytics will soon drive much of our services and systems and reshape our society.  Keeping citizens fully in the loop, keeping developers and government accountable in transparent ways are critical benefits of this action.  This is the best **'we've got your back citizens'** approach to government software that I've ever seen and I am so very proud to see this proposal surface and go forward, unflinchingly I should hope.  I give credit to the administrative executives and staff for having the prescient wisdom and courage for supporting this and all the careful work that went into in defining it. I would guess this has been under the radar of lobbyists whom otherwise would have been creating all sorts of blockades to this in Congress and tying it up with useless suits in the courts. I sure hope Congress has the smarts to jump on board, take the truly courageous leadership position, and eagerly support this initiative, one which has the potential to save the tax payer perhaps billions of dollars in just a few years--yes, even within the current election cycle and greatly increase our government's efficiency and security of operations.  Denying our government (and indirectly taxpaying citizens) the use of open source GPL software is like denying us all access to the protection of vital arms, the necessities of safe food, housing and clean water.

Amongst many carefully hidden from the public TPP elements, clearly the TPP incorrectly attempts to frame the GPL licence as invalid for commercially negotiated contracts. Ironic, when it was explicitly designed exactly to spell out terms for use of the GPL licenced software with commercial options, and the limited terms for it's re-use, including the charging of fees for the software and/or services to support it.  And the GPL (and several versions thereof) do have provisions for dual licencing of the software under other compatible licences.  Many successful public-owned and private US and other companies globally do use GPL software in it's licenced commercial form.  I hope many realize that it is Red Hat Linux (GPL licenced) which is relied upon by key agencies entrusted with the security and defense of our nation (and allies) to run millions of servers and workstations.  And we shouldn't forget the Naval vessels designed with proprietary coded Windows automation which were dead on launch due to wide-spreadd blue-screens of death and thousands of glitches, or the expensive proprietary software the FBI and IRS desparately needed to modernize which were billion dollar failures.  Social Security is also a bane to modernize and yet by having used a stable ANSI COBOL code-base has managed to limp on decades after having been coded and deployed.  The same cannot be said for applications developed using proprietary and no-longer-supported Oracle or Microsoft programming languages and deployment environments.  Microsoft's proprietary Visual Basic 5 and 6 programming environments were used by 4+ million developers to successfully write many applications for business and government usage, but cannot be used anymore since Microsoft terminated support for both the languages and deployment environment OSes, and Microsoft has designed their OS to not use them (at least not without resorting to unsupported methods). And the successor .Net languages keep getting major overhauls that make code written before require rewriting, reconfiguring and redeployment with major changes to production systems.  Some recall Microsoft's self-inflicted DLL-Hell and Typelib versioning issues, and how these were replaced with dot-Net Assembly Purgatory and now coders using Microsoft's tools have built-in Azure Cloud-dependencies, unique to Microsoft's Azure environments--which can change at it's whim and you'd better catch up immediately.  If open source software best helps the government run stable, secure, easily maintianed and auditable computerized services, then government should grab this opportunity and run with it unencumbered. To do otherwise would be guaranteeing a steady recurring of horrendous waste, fraud and expensive mistakes, perhaps critical ones which may keep American governments and citizens vulnerable to foreign intrusions and hacks.

The GPL certainly doesn't prohibit charging license fees (check the stock value of RHAT (Red Hat Linux software), even with specifying that the software source code be made available on reasonably priced media or download to the users. The GPL has already successfully been used in the US courts (see FSF (Free Software Foundation) for their victories) protecting the copyright holder(s), users  and subsequent derivations. 

The infamous failure of SCO Group to sue Linux (using IBM as the proxy) over Linux/Unix copyrights finally fell flat in US Federal courts (after dragging on needlessly and wastefully for years.) and Microsoft had been a backroom party to SCO Group's efforts to cripple GPL Gnu/Linux in all those proceedings and others.  We certainly don't need the TPP to operate as another backdoor to bring back that Zombie to life and suck more brains of citizen developers of time and efforts better spent designing, innovating, debugging, and solving the problems of government and citizens.

Whom does this sort of TPP treaty element banning GPL as commercial licence truly protect but a few multi-billion dollar US corporations each with hundreds of lawyers and lobbyists, eager to patent, re-licence as proprietary, and intimidate all other smaller smaller innovators.  It is concerning that monolithic globalized corporations such as Oracle (current owner of GPL licenced MySQL) and Microsoft continue to try to co-opt, 'embrace' for PR-effect only, then work so hard to encumber and then extinguish the open source community's millions of hours of hard work and modest profit models and work product, rather than seeking to truly benefiting the world's community of tax payers, software users and developers.  (While you're reading this, Microsoft's increasingly non-transparent update process probably tried again to foist Windows 10 upon you without knowledgeable consent, as well as more updates for security holes in its Internet Explorer & Silverlight, Adobe Flash will likely have had to put out yet another update to mitigate zero-day hacks, and Oracle's Java also had to provide yet another series of security updates.  Proprietary software bedevils the public and government with inauditable, unaccountable software that the best and brightest criminal hackers seem to enjoy hacking without consequence from anywhere on this globe (and someday the moon or Mars).

I applaud this move by the government to limit it's purchase of software creation and deployment to one-time, and allowing the rest of taxpayer funded government to benefit, that maximizes return on tax-payer dollars which should take precedence over any other claimants to the federal purse. We taxpayers do not exist for the benefit of Oracle and Microsoft and our government certainly should not be held hostage to proprietary software & hardware when viable open source and open hardware alternatives are available.  Our government already provide enormous value to these large corporations defending our nation, it's citizen (aka consumers) and enabling our way of life and freedoms, maintaining the US Patent office, Copyright Office, and court system, as well as Homeland Security and military services.

If anything, I do not want the TPP signed until revised to allow for anti-trust investigations and court proceedings to be facilitated by all signatory governments, and revised to fully support the existing case law supporting the GPL licence.  The internet (a taxpayer funded creation for our benefit) simply would not have worked for the last couple decades with out all the service-providing computers running GPL non-proprietary software. Microsoft and Oracle have been dependent upon many modules of GPL licenced and BSD-licenced code into their own software which they resell--should we take them to TPP court now for having included any of it?

It seems in this era when it is vital to be able to trust the security of vital and mission-critical government-enabling software, operating on benefit of the entire US citizenry and free world, that it is essential to have the means to audit the code, the configurations and deployments, and licences by any citizen developer competent to do so, or where necessary by any objective team of those whom meet security requirements and non-disclosure agreements, and not necessarily agents of the corporations providing the software whose vested interests and agendas may not align with the interests of tax payers.

Do this! I cannot imagine a more beneficial choice than federal agencies boldly embracing open source software and it's many wide-spread deployment options, and passing the savings and security on to all of us.

**Comment by jalbertbowden** [comment]
I attended the Federal Open Source Roundtable meeting soliciting input around the administration's commitment to publish an open source software policy laid out in the Second Open Government National Action Plan (NAP) this past November, and the entire time the TPP was in the back of my mind. Consideration was not given vocally at the time and for that I apologize sincerely. I spent the better part of the meeting listening to lobbyists argue that open source licenses would kill their clients' profits margins, an opinion that is expected, but not what I expected at this meeting. 
The entire time I spent listening and arguing, in the back of my mind I kept thinking "this is all m00t if tpp passes and is enforced". Again, I apologize to the world for that. I should have done better.
